prime-router/src/main/kotlin/fhirengine/utils/FHIRBundleHelpers.kt

```diff
@@ -80,7 +80,8 @@ object FHIRBundleHelpers {
         val provenanceResource = try {
```

Comment:
Again, depends on if this is just for connectathon or if it'll eventually be merged into master, but this function is kind of doing two things, adding a provenance and adding references to reports within the provenance, it might be nice to split out the piece that does the try/catch since that's currently happening twice in this path. Not essential, especially if this is just for connectathon and won't be merged into master.

Reply:
The catch part of the try/catch is different though. In the addProvenanceReference method it is creating the provenance. In the addReceivers method it calls the addProvenanceReference because we not only want to create the provenance, we want to populate it. However, we do not want to call that method if the provenance already exists. Does that make sense?

Reply:
Yea it does, do we only want/need to populate it if we're creating it for the first time, or should we maybe just be calling `addProvenanceReference` either way?

Reply:
The second call is in the addProvenanceReference method, so it would not make sense for it to call itself.

Reply:
Sorry confusing comment, I'm going to resolve this thread and comment at the piece of code I'm talking about.